src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -293,31 +296,54 @@ class PeerActor(
       }
   }
 
-  class HandshakedHandler(rlpxConnection: RLPxConnection, initialStatus: msg.Status) {
+  class HandshakedHandler(rlpxConnection: RLPxConnection, initialStatus: msg.Status, currentPeerMaxBlock: BigInt) {
 
     def receive: Receive =
       handleSubscriptions orElse handleTerminated(rlpxConnection) orElse
       handlePeerChainCheck(rlpxConnection) orElse handlePingMsg(rlpxConnection) orElse
       handleBlockFastDownload(rlpxConnection) orElse {
       case RLPxConnectionHandler.MessageReceived(message) =>
         log.debug("Received message: {}", message)
+        updateMaxBlock(message)
         notifySubscribers(message)
         processMessage(message)
 
       case s: SendMessage[_] =>
+        updateMaxBlock(s.message)
         rlpxConnection.sendMessage(s.message)(s.enc)
 
+      case GetMaxBlockNumber(actor) => actor ! MaxBlockNumber(currentPeerMaxBlock)
```

Comment:
we can do it with the sender, I added explicit reference to ensure that it is specified because non-actor can also send message to actor